### PR TITLE
Do not set controller link-layer data out-of-band

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch accompanies #11683.

It removes the logic whereby the machine agent, if running on a controller, sets link-layer device information from the provider before starting the various workers.

There is no need for this step to be performed. The machiner worker will populate device information as soon as it starts, then the instance-poller will decorate this with provider data.

In the case of AWS, it fixes a bug whereby we had duplicated devices due to device names being unsupported by the provider, and filled with a placeholder such as _unsupported0_ by Juju.

This was the only usage of `SetProviderNetworkConfig` from the common networking API. It is now removed, and that method stubbed for old versions of the `machiner` API, which embeds it. Stubbing it (allowing removal of the common logic) is safe - for new bootstraps, QA verifies this. For upgrades, the controller machine agent has already used it to set link-layer data.

## QA steps

On a variety of substrates:
- Bootstrap and deploy a workload; check that it settles without error. 
- ```juju enable-ha``` and check that the controllers arrive at a healthy state.
- Connect to Mongo and see that the `linklayerdevices` collection has the devices we expect.

I tried LXD, MAAS, AWS.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1855263

### CODE REVIEW

File: cmd/jujud/agent/machine.go [hunk 1]
```diff
@@ -728,14 +730,7 @@ func (a *MachineAgent) startAPIWorkers(apiConn api.Connection) (_ worker.Worker,
 			// the API, report "unknown job type" here.
 		}
 	}
-	if isController {
```
Comment (manadart) [inline cmd/jujud/agent/machine.go:731]: The removal of this block and the method it calls are the material changes here.
The rest is drive-by tidy-up.

**Comment by hmlanigan** [comment]
!!build!!

File: api/facadeversions.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ var facadeVersions = map[string]int{
 	"MachineActions":               1,
 	"MachineManager":               6,
 	"MachineUndertaker":            1,
-	"Machiner":                     3,
+	"Machiner":                     4,
```
Comment (hmlanigan) [inline api/facadeversions.go:69]: Usually we don't update a facade version in a dot release.  Do we have approval for this?  Can we delay that part to develop only?

I'm all for removing code, but not sure it's a good enough reason to bump a facade version in a dot release.   And having to deal with potential versioning fall out

Comment (wallyworld) [reply api/facadeversions.go:69]: I think in this case it's ok. It's a controller facade and there will be no issues therefore with version skew. It matters a lot more if it were a client facade, as then the version X CLI would not fully know about a newer X=1 facade in a newer controller.

**Comment by manadart** [comment]
$$merge$$
